app/src/core/database_manager.py

```
@@ -0,0 +1,253 @@
+                # Simple validation query to ensure connection is alive
+                with conn.cursor() as cursor:
+                    cursor.execute("SELECT 1")
+                logger.debug("Successfully retrieved a valid database connection.")
+                return conn
+            except OperationalError as e:
+                if conn:
+                    try:
+                        # Return and close the broken connection to avoid leaks
+                        self._pool.putconn(conn, close=True)
+                    except Exception:
+                        pass
```

Ruff (0.12.2)

Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)